File: core/models/models/node.go
package models

import (
	"time"
)

type Node struct {
	any            `collection:"nodes"`
	BaseModel      `bson:",inline"`
	Key            string    `json:"key" bson:"key" description:"Key"`
	Name           string    `json:"name" bson:"name" description:"Name"`
	Ip             string    `json:"ip" bson:"ip" description:"IP"`
	Mac            string    `json:"mac" bson:"mac" description:"MAC"`
	Hostname       string    `json:"hostname" bson:"hostname" description:"Hostname"`
	Description    string    `json:"description" bson:"description" description:"Description"`
	IsMaster       bool      `json:"is_master" bson:"is_master" description:"Is master"`
	Status         string    `json:"status" bson:"status" description:"Status"`
	Enabled        bool      `json:"enabled" bson:"enabled" description:"Enabled"`
	Active         bool      `json:"active" bson:"active" description:"Active"`
	ActiveAt       time.Time `json:"active_at" bson:"active_ts" description:"Active at"`
	CurrentRunners int       `json:"current_runners" bson:"current_runners" description:"Current runners"`
	MaxRunners     int       `json:"max_runners" bson:"max_runners" description:"Max runners"`
}

File: frontend/crawlab-ui/src/i18n/lang/en/components/node.ts
const node: LComponentsNode = {
  form: {
    key: 'Unique Identity Key',
    name: 'Name',
    type: 'Type',
    ip: 'IP',
    mac: 'MAC Address',
    hostname: 'Hostname',
    enabled: 'Enabled',
    maxRunners: 'Max Runners',
    description: 'Description',
    status: 'Status',
  },
  nodeType: {
    label: {
      master: 'Master',
      worker: 'Worker',
    },
  },
  nodeStatus: {
    label: {
      online: 'Online',
      offline: 'Offline',
      unknown: 'Unknown',
    },
    tooltip: {
      unregistered: 'Node is waiting to be registered',
      registered: 'Node is registered and wait to be online',
      online: 'Node is currently online',
      offline: 'Node is currently offline',
      unknown: 'Unknown node status',
    },
  },
  nodeRunners: {
    tooltip: {
      unavailable: 'No runners available at this moment',
      running: '{running} out of {max} runners are running',
      available: 'All runners available',
    },
  },
};

export default node;
